### PR TITLE
Maximal rank and full rank row submatrices

### PR DESCRIPTION
**Pull request by CohenCyril**
##### Motivation for this change

Taking a submatrix of maximal rank, and even a square matrix of full
rank when given a proof that the rank was already full.
Part of #207

##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)
- [x] added corresponding documentation in the headers
- [x] merge of #574

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

### CODE REVIEW

**Comment by CohenCyril** [comment]
@Boutry @thery do you think there is a chance you have the time to do a review before the release?

**Comment by Boutry** [comment]
> @Boutry @thery do you think there is a chance you have the time to do a review before the release?

I will try to do it by Monday. Do you have examples of reviews? I checked some of the closed PR and there were few comments. So I am still a bit unclear about how a review should look like. (I looked at the checklist though.)

**Comment by CohenCyril** [comment]
I guess you could take example from similar PRs (some went through as they were, others required more work):
- https://github.com/math-comp/math-comp/pull/618
- https://github.com/math-comp/math-comp/pull/585
- https://github.com/math-comp/math-comp/pull/574
- https://github.com/math-comp/math-comp/pull/558

**Comment by Boutry** [comment]
> I guess you could take example from similar PRs (some went through as they were, others required more work):
> 
> * #618
> * #585
> * #574
> * #558

Thank you for the links. I believe line 284 in the Changelog belongs to another PR. Aside from that everything looks good to me.

**Comment by CohenCyril** [comment]
@Boutry I attended the issue, thanks for pointing it out.

Please click on "Files changed" , then "Review changes", then type a review and click on "Approve", or if you changed your mind "Comment" or "Request Changes". Then ask a core developer to merge if you accepted, by tagging @math-comp/core

**Review by Boutry — APPROVED** [review]

**Comment by Boutry** [comment]
Doesn't Laurent need to approve it as well?

**Comment by CohenCyril** [comment]
> Doesn't Laurent need to approve it as well?

Ask @ybertot

**Comment by Boutry** [comment]
He just called me and seems fine with just my approval. Thank you for the help!

**Comment by CohenCyril** [comment]
@math-comp/core when CI passes, I guess you can merge.

**Comment by Boutry** [comment]
It is now the case.

**Comment by gares** [comment]
I don't get exactly how this merge process works, but I guess che posso premere il bottone

**Pull request merged**
